diff --git a/lib/middleware/karma.js b/lib/middleware/karma.js
index c48332d12c644318cc6b832469dc8ecda327700b..72e1735a1d45fe1ac46ebc8917b54512db27fdf9 100644
--- a/lib/middleware/karma.js
+++ b/lib/middleware/karma.js
@@ -220,11 +220,11 @@ function createKarmaMiddleware (
           }) : []

           return data
-            .replace('%SCRIPTS%', scriptTags.join('\n'))
-            .replace('%CLIENT_CONFIG%', 'window.__karma__.config = ' + JSON.stringify(client) + ';\n')
-            .replace('%SCRIPT_URL_ARRAY%', 'window.__karma__.scriptUrls = ' + JSON.stringify(scriptUrls) + ';\n')
-            .replace('%MAPPINGS%', 'window.__karma__.files = {\n' + mappings.join(',\n') + '\n};\n')
-            .replace('\n%X_UA_COMPATIBLE%', getXUACompatibleMetaElement(request.url))
+            .replace('%SCRIPTS%', () => scriptTags.join('\n'))
+            .replace('%CLIENT_CONFIG%', () => 'window.__karma__.config = ' + JSON.stringify(client) + ';\n')
+            .replace('%SCRIPT_URL_ARRAY%', () => 'window.__karma__.scriptUrls = ' + JSON.stringify(scriptUrls) + ';\n')
+            .replace('%MAPPINGS%', () => 'window.__karma__.files = {\n' + mappings.join(',\n') + '\n};\n')
+            .replace('\n%X_UA_COMPATIBLE%', () => getXUACompatibleMetaElement(request.url))
         })
       })
     } else if (requestUrl === '/context.json') {
